tests/vov-bug-finder/test_view_of_views_bug_finder.cpp

```diff
@@ -0,0 +1,127 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include "Kokkos_Core.hpp"
+#include "gtest/gtest.h"
+
+void test_no_throw_placement_new_in_parallel_for() {
+  ASSERT_NO_THROW(({
+    using V = Kokkos::View<int *>;
+    Kokkos::View<V **, Kokkos::HostSpace> vov(
+        Kokkos::view_alloc("vov", Kokkos::WithoutInitializing), 2, 3);
+    V a("a", 4);
+    V b("b", 5);
+    Kokkos::parallel_for(
+        "Fine", Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(0, 1),
+        KOKKOS_LAMBDA(int) {
+          new (&vov(0, 0)) V(a);
+          new (&vov(0, 1)) V(a);
+          new (&vov(1, 0)) V(b);
+        });
+  }));
+}
+
+void test_death_allocation_in_parallel_for() {
+  ASSERT_DEATH(
+      ({
+        using V = Kokkos::View<int *>;
+        Kokkos::View<V **, Kokkos::HostSpace> vov(
+            Kokkos::view_alloc("vov", Kokkos::WithoutInitializing), 2, 3);
+        V a("a", 4);
+        new (&vov(0, 0)) V(a);
+        new (&vov(0, 1)) V(a);
+        Kokkos::parallel_for(
+            "AllocatesInParallel]For",
+            Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(0, 1),
+            KOKKOS_LAMBDA(int) {
+              V b("b", 5);
```

Comment:
In a `Cuda` build, the compiler is complaining about this line that:

```
warning #20011-D: calling a __host__  function ... is not allowed.
```

The reason seems to be that we end up in a constructor that isn't marked with Kokkos markup. One option would be to desactivate this part of the test in builds with a device backend. I'm not sure whether there's a better solution to this problem.